Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/positive/SubmissionTestResultKeysSharedFragment.kt

```diff
@@ -75,8 +75,13 @@ class SubmissionTestResultKeysSharedFragment :
             showMoveToRecycleBinDialog()
         }
 
-        viewModel.routeToScreen.observe2(this) { navDirections ->
-            navDirections?.let { doNavigate(it) } ?: popBackStack()
+        viewModel.routeToScreen.observe2(this) {
+            if (navArgs.comesFromDispatcherFragment) {
+                doNavigate(
+                    SubmissionTestResultKeysSharedFragmentDirections
+                        .actionGlobalToSubmissionTestResultKeysSharedFragmentToMainFragment()
+                )
```

Comment:
I think the first To does not belong there, meaning it should be actionGlobalSubmissionTestResultKeysSharedFragmentToMainFragment

Reply:
Yes, good catch, it should be just `actionSubmissionTestResultKeysSharedFragmentToMainFragment`